runtime/lua/vim/diagnostic.lua

```diff
@@ -755,7 +755,31 @@ function M.get(bufnr, opts)
     opts = { opts, 't', true },
   })
 
-  return get_diagnostics(bufnr, opts, false)
+  return vim.deepcopy(get_diagnostics(bufnr, opts, false))
+end
+
+--- Get current diagnostics count.
+---
+---@param bufnr integer|nil Buffer number to get diagnostics from. Use 0 for
+---                        current buffer or nil for all buffers.
+---@param opts table|nil A table with the following keys:
+---                        - namespace: (number) Limit diagnostics to the given namespace.
+---                        - lnum: (number) Limit diagnostics to the given line number.
+---                        - severity: See |diagnostic-severity|.
+---@return table A table with actually present severity counts.
```

Comment:
This could be more elaborated (like explaining what keys are), or with more concrete examples.

Reply:
They're the same as in `vim.diagnostic.get()`, from where this is copied. _That_ would be the place for improvements.

Reply:
I think this was more about the return table.

So something like `A table with actually present severity values as keys (see |diagnostic-severity|) and integer counts as values.`?